impeller/entity/save_layer_utils.cc

```diff
@@ -106,8 +107,34 @@ std::optional<Rect> ComputeSaveLayerCoverage(
 
   // Transform the input coverage into the global coordinate space before
   // computing the bounds limit intersection.
-  return coverage.TransformBounds(effect_transform)
-      .Intersection(coverage_limit);
+  Rect transformed_coverage = coverage.TransformBounds(effect_transform);
+  std::optional<Rect> intersection =
+      transformed_coverage.Intersection(coverage_limit);
+  if (!intersection.has_value()) {
+    return std::nullopt;
+  }
+
+  // Sometimes a saveLayer is only slightly shifted outside of the cull rect,
+  // but is being animated in. This is common for the Android slide in page
+  // transitions. In these cases, computing a cull that is too tight can cause
+  // thrasing of the texture cache. Instead, we try to determine the
+  // intersection using only the sizing by shifting the coverage rect into the
+  // cull rect origin.
+  Point delta = coverage_limit.GetOrigin() - transformed_coverage.GetOrigin();
+  if (ScalarNearlyEqual(delta.y, 0) || ScalarNearlyEqual(delta.x, 0)) {
+    Scalar threshold = std::max(std::abs(delta.x / coverage_limit.GetWidth()),
+                                std::abs(delta.y / coverage_limit.GetHeight()));
+    if (threshold < 0.3) {
```

Comment:
This magic number needs to be pulled out into a const with documentation about where it comes from.

Reply:
done